crates/l2/tests/tests.rs

```diff
@@ -396,6 +399,93 @@ async fn test_erc20_roundtrip(
     Ok(())
 }
 
+async fn test_forced_withdrawal(
+    l2_client: &EthClient,
+    l1_client: &EthClient,
+) -> Result<(), Box<dyn std::error::Error>> {
+    let rich_wallet_private_key = l1_rich_wallet_private_key();
+    let rich_address = ethrex_l2_sdk::get_address_from_secret_key(&rich_wallet_private_key)
+        .expect("Failed to get address");
+    let l1_initial_balance = l1_client
+        .get_balance(rich_address, BlockIdentifier::Tag(BlockTag::Latest))
+        .await?;
+    let transfer_value = U256::from(100);
+    let mut l1_gas_costs = 0;
+
+    let calldata = encode_calldata("withdraw(address)", &[Value::Address(rich_address)])?;
+
+    let l1_to_l2_tx_hash = ethrex_l2_sdk::send_l1_to_l2_tx(
+        rich_address,
+        Some(0),
+        Some(21000 * 10),
+        L1ToL2TransactionData::new(
+            COMMON_BRIDGE_L2_ADDRESS,
+            21000 * 5,
+            transfer_value,
+            Bytes::from(calldata),
+        ),
+        &l1_rich_wallet_private_key(),
+        common_bridge_address(),
+        l1_client,
+    )
+    .await?;
+
+    println!("Waiting for L1 to L2 transaction receipt on L1");
+
+    let l1_to_l2_tx_receipt = wait_for_transaction_receipt(l1_to_l2_tx_hash, l1_client, 5).await?;
+    l1_gas_costs +=
+        l1_to_l2_tx_receipt.tx_info.gas_used * l1_to_l2_tx_receipt.tx_info.effective_gas_price;
+    println!("Waiting for L1 to L2 transaction receipt on L2");
+
+    let res = wait_for_l2_deposit_receipt(
+        l1_to_l2_tx_receipt.block_info.block_number,
+        l1_client,
+        l2_client,
+    )
+    .await?;
+
+    let proof = l2_client
+        .wait_for_message_proof(res.tx_info.transaction_hash, 1000)
+        .await;
+    let proof = proof.unwrap().into_iter().next().expect("proof not found");
+
+    let on_chain_proposer_address = Address::from_str(
+        &std::env::var("ETHREX_COMMITTER_ON_CHAIN_PROPOSER_ADDRESS")
+            .expect("ETHREX_COMMITTER_ON_CHAIN_PROPOSER_ADDRESS env var not set"),
+    )
+    .unwrap();
+    while l1_client
+        .get_last_verified_batch(on_chain_proposer_address)
+        .await
+        .unwrap()
+        < proof.batch_number
+    {
+        println!("Withdrawal is not verified on L1 yet");
+        tokio::time::sleep(Duration::from_secs(2)).await;
+    }
+
+    let withdraw_claim_tx = claim_withdraw(
+        transfer_value,
+        rich_address,
+        rich_wallet_private_key,
+        l1_client,
+        &proof,
+    )
+    .await
+    .expect("error while claiming");
+    let res = wait_for_transaction_receipt(withdraw_claim_tx, l1_client, 5).await?;
+    l1_gas_costs += res.tx_info.gas_used * res.tx_info.effective_gas_price;
+
+    let l1_final_balance = l1_client
+        .get_balance(rich_address, BlockIdentifier::Tag(BlockTag::Latest))
+        .await?;
+    assert_eq!(
+        l1_initial_balance + transfer_value - l1_gas_costs,
+        l1_final_balance
+    );
```

Comment:
Here we should also ensure that the l2 balance of the account is the one we expect it to be (i.e. just decreased by the withdrawal value since the gas cost was paid on L1)